open-craft/opencraft
# Add login flow

#### Testing instructions

1. Checkout this branch
2. Run migrations
3. `make rundev`
4. Go to `/`. Should redirect to `/registration/`
5. Fill in and submit registration form
6. You should be redirected back to the registration form. Verify that:
  - You are logged in as the new user
    - Email address and 'logout' link appear at the top right
    - Session cookie set
  - There is an 'application successfully sent' message
  - There is a 'pending email confirmation' message
  - The form is populated
  - The 'Full name', 'Your project' and 'Subscribe to updates' fields are editable, all other fields are read only
  - Read only fields cannot be modified by bypassing the form and sending a manual post request
7. Click the 'logout' link at the top right. Verify that you are logged out
8. Click the 'login' button at the top right. Verify that client-side validation of login form works
9. Login with the username + password you registered with in step 5. Verify that you see the same form as in step 6
10. Logout and click 'login' again
11. Login with the email + password you registered with in step 5. You should be able to log in with either username or email address
12. Search through the `make rundev` console output for the email verification emails. There should be one email sent to the user's email address, and another to the 'public contact email' address (or only one email if the same address was used for both). Click the links in both emails
13. Refresh the registration form. Verify that the 'pending email confirmation' message disappears
